HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -156,6 +201,43 @@ void HLTScoutingEgammaProducer::produce(edm::StreamID sid, edm::Event& iEvent, e
     reco::SuperClusterRef scRef = candidate.superCluster();
     if (scRef.isNull() && !scRef.isAvailable())
       continue;
+
+    reco::CaloClusterPtr SCseed = candidate.superCluster()->seed();
+    const EcalRecHitCollection* rechits = (std::abs(scRef->eta()) < 1.479) ? rechitsEB.product() : rechitsEE.product();
+    Cluster2ndMoments moments = EcalClusterTools::cluster2ndMoments(*SCseed, *rechits);
+    float sMin = moments.sMin;
+    float sMaj = moments.sMaj;
+
+    unsigned int seedId = (*SCseed).seed();
+
+    std::vector<DetId> mDetIds = EcalClusterTools::matrixDetId((topology), (*SCseed).seed(), rechitMatrixSize);
+    std::vector<float> mEnergies;
+    std::vector<float> mTimes;
+
+    int detSize = mDetIds.size();
+
+    if (!saveRecHitTiming)
+      mTimes.push_back(0.);
+
+    for (int i = 0; i < pow(2 * rechitMatrixSize + 1, 2); i++) {
```

Comment:
The loop goes from `0` to `(2 * rechitMatrixSize + 1)² - 1` (included).
Is it the intended range ?

Reply:
How large is this ?
It is probably faster to
  - allocate the vectors with the correct size up front
  - initialise them to all `0`
  - only fill the valid elements

rather than incrementing the size one element at a time.

E.g.
```c++
size = (2 * rechitMatrixSize + 1) * (2 * rechitMatrixSize + 1);  // if it's correct
std::vector<float> mEnergies(size, 0.);
std::vector<float> mTimes(size, 0.);

size = std:min(size, detSize);
for (int i = 0; i < size, 2); i++) {
  mEnergies[i] = MiniFloatConverter::reduceMantissaToNbitsRounding(recHitE(mDetIds.at(i), *rechits), mantissaPrecision));
  if (saveRecHitTiming)
    mTimes[i] = MiniFloatConverter::reduceMantissaToNbitsRounding(recHitT(mDetIds.at(i), *rechits), mantissaPrecision));
}
```

Reply:
However, why do you store `(2 * rechitMatrixSize + 1)²` elements, if only those up to `detSize` are valid ?

Reply:
Thanks. This implementation is indeed not good, I will commit a better version in the next couple of hours.

Reply:
this should be resolved now.